cmd/kubeless/autoscaleCreate.go

```diff
@@ -28,33 +41,36 @@ var autoscaleCreateCmd = &cobra.Command{
 		} else if max < min {
 			logrus.Fatalf("max must be greater than or equal to min")
 		}
-		ns, err := cmd.Flags().GetString("namespace")
-		if err != nil {
-			logrus.Fatal(err.Error())
-		}
-		if ns == "" {
-			ns = utils.GetDefaultNamespace()
-		}
 
 		metric, err := cmd.Flags().GetString("metric")
 		if err != nil {
-			logrus.Fatal(err.Error())
+			logrus.Fatal(err)
 		}
 		if metric != "cpu" && metric != "qps" {
 			logrus.Fatalf("only supported metrics: cpu, qps")
 		}
 
 		value, err := cmd.Flags().GetString("value")
 		if err != nil {
-			logrus.Fatal(err.Error())
+			logrus.Fatal(err)
 		}
 
-		client := utils.GetClientOutOfCluster()
+		hpa, err := utils.GetHorizontalAutoscaleDefinition(funcName, ns, metric, min, max, value)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+		function.Spec.HorizontalPodAutoscaler = hpa
 
-		err = utils.CreateAutoscale(client, funcName, ns, metric, min, max, value)
+		crdClient, err := utils.GetCRDClientOutOfCluster()
+		if err != nil {
+			logrus.Fatal(err)
+		}
+		logrus.Infof("Redeploying function with autoscaling parameters...")
```

Comment:
This message might not true. The function itself isn't redeployed if I understand this PR correctly. The controller will be triggered with changes from the CRD object to create HPA object. So it's better to say: "adding autoscaling rule to the function..."